### PR TITLE
Implementation of Finchley / Boot 2 migration

### LINKED ISSUE #40: Add Support for Spring Boot 2.x.x
Spring Boot 2 has changed a lot of the core framework of spring boot, including several things with the autowire process.

As a result, these connectors don't work in a spring boot 2 application as soon as it is deployed to PCF.

(Specifically, the Eureka binder crashes with a class not found exception because the relaxed property resolver been removed/moved/changed)

Exception:

Caused by: org.springframework.beans.factory.UnsatisfiedDependencyException: Error creating bean with name 'discoveryCompositeHealthIndicator' defined in class path resource [org/springframework/cloud/client/CommonsClientAutoConfiguration$DiscoveryLoadBalancerConfiguration.class]: Unsatisfied dependency expressed through method 'discoveryCompositeHealthIndicator' parameter 1; nested exception is org.springframework.beans.factory.UnsatisfiedDependencyException: Error creating bean with name 'eurekaHealthIndicator' defined in class path resource [org/springframework/cloud/netflix/eureka/EurekaClientAutoConfiguration$EurekaHealthIndicatorConfiguration.class]: Unsatisfied dependency expressed through method 'eurekaHealthIndicator' parameter 1; nested exception is org.springframework.beans.factory.BeanCreationException: Error creating bean with name 'eurekaInstanceConfigBean' defined in class path resource [io/pivotal/spring/cloud/service/eureka/EurekaInstanceAutoConfiguration.class]: Initialization of bean failed; nested exception is java.lang.NoClassDefFoundError: org/springframework/boot/bind/RelaxedPropertyResolver

this was produced with:
Spring Boot 2.0.0.RELEASE
And version 1.6.1.RELEASE of this repository's discovery client library

Ultimately, the Apex library is from this repo https://github.com/pivotal-cf/spring-cloud-services-starters for the spring-cloud-services-starter-service-registry project, however I have a hunch that the code update is needed in this project.  This problem probably impacts all the spring cloud services starters for PCF, I've just only tested it with Eureka.

### PR DESCRIPTION
**Pull request by ojhughes**
Add spring MVC configuration in plain text config server test that allows for default label to by the client instead of specifying a label
Use boot 2 binding API instead of relaxed property binder
Rework vault property master to use boot property `keys-to-sanitize`
instead of custom solution.

resolves #40

### CODE REVIEW

File: gradle/wrapper/gradle-wrapper.properties [hunk 1]
```diff
@@ -1,6 +1,5 @@
-#Thu Jan 04 14:46:10 GMT 2018
 distributionBase=GRADLE_USER_HOME
 distributionPath=wrapper/dists
 zipStoreBase=GRADLE_USER_HOME
 zipStorePath=wrapper/dists
-distributionUrl=https\://services.gradle.org/distributions/gradle-4.4.1-all.zip
+distributionUrl=https\://services.gradle.org/distributions/gradle-4.4.1-bin.zip
```
Comment (spikymonkey) [inline gradle/wrapper/gradle-wrapper.properties]: Please could you leave this with the `-all` distribution? It gives IntelliJ access to Gradle API info, which can be useful when working on the build scripts.

File: spring-cloud-services-spring-connector/src/main/java/io/pivotal/spring/cloud/service/config/PlainTextOAuth2ConfigClient.java [hunk 2]
```diff
@@ -89,9 +90,12 @@ public Resource getConfigFile(String profile, String label, String path) {
 			label = configClientProperties.getLabel();
 		}
 
-		return restTemplate.getForEntity(configClientProperties.getUri() + "/"
+		String url = configClientProperties.getUri() + "/"
 				+ configClientProperties.getName() + "/" + profile + "/"
-				+ (label == null ? path + "?useDefaultLabel" : label + "/" + path),
-				Resource.class).getBody();
+				+ (label == null ? path + "?useDefaultLabel" : label + "/" + path);
+		System.out.println(url);
```
Comment (spikymonkey) [inline spring-cloud-services-spring-connector/src/main/java/io/pivotal/spring/cloud/service/config/PlainTextOAuth2ConfigClient.java]: Debugging `println` that should be removed?

File: spring-cloud-services-spring-connector/src/test/java/io/pivotal/spring/cloud/service/config/client/server/ConfigServerTestApplication.java [hunk 3]
```diff
@@ -50,29 +50,35 @@
 @RestController
 @EnableConfigServer
 @SpringBootApplication(exclude = { RabbitAutoConfiguration.class,
-		ServoMetricsAutoConfiguration.class })
+//		ServoMetricsAutoConfiguration.class
```
Comment (spikymonkey) [inline spring-cloud-services-spring-connector/src/test/java/io/pivotal/spring/cloud/service/config/client/server/ConfigServerTestApplication.java]: Commented out code should be deleted

File: spring-cloud-services-spring-connector/src/test/java/io/pivotal/spring/cloud/service/eureka/SanitizingEurekaInstanceConfigBeanTest.java [hunk 4]
```diff
@@ -134,12 +138,11 @@ public void testExceptionThrownIfSvhnDiffersFromAppName() {
...
 		this.ctx = new AnnotationConfigApplicationContext();
 
-		ArrayList<String> pairs1 = new ArrayList();
-
-		for (String pair : pairs) {
-			pairs1.add(pair);
-		}
+		List<String> pairs1 = new ArrayList<>(Arrays.asList(pairs));
 		pairs1.add("sanitizingEurekaInstanceConfigBean.integration.test:true");
+//		pairs1.add("spring.application.name:testapp1");
```
Comment (spikymonkey) [inline spring-cloud-services-spring-connector/src/test/java/io/pivotal/spring/cloud/service/eureka/SanitizingEurekaInstanceConfigBeanTest.java]: Commented out code should be deleted

File: build.gradle [hunk 5]
```diff
@@ -4,6 +4,11 @@ buildscript {
 		mavenCentral()
 	}
 
+	ext.springBootVersion = "2.0.0.RELEASE"
+	ext.springCloudVersion = "Finchley.M9"
+	ext.lombokVersion = "1.16.8"
+	ext.springCloudNetflixVersion = "2.0.0.M7"
```
Comment (spikymonkey) [inline build.gradle]: Wondered if we should explicitly set the spring cloud netflix version like this, or import the Finchley BOM and have it dependency managed for us? Finchley.M9 specifies SCN 2.0.0.M8, so we're already behind here.

Comment (ojhughes) [reply build.gradle]: Nice spot, removed that line as it wasn't used

File: gradlew [hunk 6]
```diff
@@ -1,4 +1,4 @@
-#!/usr/bin/env bash
+#!/usr/bin/env sh
```
Comment (spikymonkey) [inline gradlew]: I think this file could probably be reverted too?

File: gradlew.bat [hunk 7]
```diff
@@ -49,7 +49,6 @@ goto fail
 @rem Get command-line arguments, handling Windows variants
 
 if not "%OS%" == "Windows_NT" goto win9xME_args
-if "%@eval[2+2]" == "4" goto 4NT_args
```
Comment (spikymonkey) [inline gradlew.bat]: ...and this one

File: gradlew (hunk 6 above)
Comment (ojhughes) [reply gradlew]: Yep think this is the work of `gradle wrapper` command

Comment (spikymonkey) [reply gradlew]: Sorry, I meant I think this whole file should be reverted. We're not bumping gradle or anything as part of this change, so I don't think the gradlew script needs to change at all.

**Comment by ojhughes** [comment]
Thanks for review @spikymonkey let me know if anything else

**Pull request merged**